### ISSUE TITLE
GLRecord with Steam: `GLIBCXX_3.4.20' not found

### ISSUE THREAD

**Issue by Lucki** [MaartenBaert/ssr#214]
I've tried to capture a game through steam with `ssr-glinject %command%`. Commit 2c71d72 already applied.
Steam gives me this error message: `/bin/bash: /home/lucki/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version 'GLIBCXX_3.4.20' not found (required by /usr/lib/libssr-glinject.so)`.

**Comment by Brottweiler** [comment]
Where exactly did you copy `libssr-glinject.so` to?

I tried copying it to `.local/share/Steam/ubuntu12_32/steam-runtime` and `.local/share/Steam/ubuntu12_32/steam-runtime/i386/lib/i386-linux-gnu` but neither makes a difference.

**Comment by Brottweiler** [comment]
Also I'd like to ask something on this topic.

I added Minecraft to Steam. I added the `ssr-glinject %command%` launch option. Started SSR, OpenGL settings is empty except `channel-brott` as Channel name (default, haven't messed with it).

I go to the recording screen, then launch minecraft. But I cannot record. 

I am getting the usual message when you don't have an OpenGL even opened:

```
[PageRecord::StartOutput] Error: Could not get the size of the OpenGL application. Either the application wasn't started correctly, or the application hasn't created an OpenGL window yet. If you want to start recording before starting the application, you have to enable scaling and enter the video size manually.
```

Any idea how to make this work?

**Comment by MaartenBaert** [comment]
@Brottweiler Start steam from a terminal so you can see the error messages from ssr-glinject. Also, the correct path is ~/.steam/steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu.

**Comment by Brottweiler** [comment]
I placed `libssr-glinject.so` into `.local/share/Steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu` but I still get the error OP gets:

```
Game update: AppID 300 "Day of Defeat: Source", ProcID 29328, IP 0.0.0.0:0
ERROR: ld.so: object '/home/brott/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/brott/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ssr-glinject: LD_PRELOAD = :/home/brott/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so:/home/brott/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so:libssr-glinject.so
ssr-glinject: command = /home/brott/.local/share/Steam/steamapps/common/Day of Defeat Source/hl2.sh -game dod -steam
ERROR: ld.so: object '/home/brott/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
/bin/bash: /home/brott/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /usr/lib/libssr-glinject.so)
Game removed: AppID 300 "Day of Defeat: Source", ProcID 29328 
```

**Comment by MaartenBaert** [comment]
Look at the last line: ```(required by /usr/lib/libssr-glinject.so)```
The loader is still trying to use the system libssr-glinject and not the one for Ubuntu. Are you absolutely sure that you have placed the right version in the right folder?

To be clear, that means:
* Put the 32-bit Ubuntu libssr-glinject.so in ```~/.steam/steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu```.
* Put the 64-bit Ubuntu libssr-glinject.so in ```~/.steam/steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu```.
I suspect you missed the 64-bit one, or you put the 32-bit library in the 64-bit folder. In any case the loader hasn't found the 64-bit Ubuntu libssr-glinject, that's why it doesn't work.

**Comment by Brottweiler** [comment]
You are correct. I only copied the i386 one, and not the amd64 one. Now when I copied over the 2nd one, it works! I think I misunderstood at first.

**Comment by xpris** [comment]
HI.

I have still this issues:
I copy two libs like you and not working on my Ubuntu 15.15 x64 but work on Mageia 5 x64.
On Mageia5 i use ssr 0.3.3 (working fine) and on Ubuntu I use ssr 0.3.6 (not working).

In Ubuntu I have error:

Game update: AppID 219740 "Don't Starve", ProcID 6833, IP 0.0.0.0:0
ERROR: ld.so: object '/home/x/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/x/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ssr-glinject: LD_PRELOAD = :/home/x/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so:/home/x/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so:libssr-glinject.so
ssr-glinject: command = /media/x/ACER-DANE/steamapps/common/dont_starve/bin/dontstarve_steam
ERROR: ld.so: object '/home/x/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
Installing breakpad exception handler for appid(steam)/version(1447125378)
Installing breakpad exception handler for appid(steam)/version(1447125378)
/media/x/ACER-DANE/steamapps/common/dont_starve/bin/dontstarve_steam: /home/x/.local/share/Steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /home/x/.local/share/Steam/ubuntu12_32/steam-runtime/i386/lib/i386-linux-gnu/libssr-glinject.so)
/media/x/ACER-DANE/steamapps/common/dont_starve/bin/dontstarve_steam: /home/x/.local/share/Steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /home/x/.local/share/Steam/ubuntu12_32/steam-runtime/i386/lib/i386-linux-gnu/libssr-glinject.so)
Game removed: AppID 219740 "Don't Starve", ProcID 6833 


So whats is going on? I copy this libs, so what?

**Comment by MaartenBaert** [comment]
Make sure that you are using the libraries from the Ubuntu 12.04 (precise) package that match the version of SSR you are using. You should always use the 12.04 libraries for the Steam runtime since Steam is based on 12.04. You can get the latest packages here:
https://launchpad.net/~maarten-baert/+archive/ubuntu/simplescreenrecorder/+packages

**Comment by xpris** [comment]
You have right. I have libs from ssr 0.3.6 but from Ubuntu 15.10. Helped swaping to lib ssr 0.3.6 from Ubuntu 12.04.

Thanks.

**Comment by MaartenBaert** [comment]
Just a bugfix. I will update the changelog.

**Comment by aramboi** [comment]
I created a simple shell script to automate this fix https://gist.github.com/aramboi/a4868861797d243778c0
You can run thin after the Steam runtime is updated and overrides the overrides. :)
